### PR TITLE
Add use of auto keep-alive

### PR DESCRIPTION
**Pull request by mdmarek**
Add use of auto keep-alive, also allow per server or client logger.

### CODE REVIEW

File: grid.v3/cfg.go [hunk 1]
```diff
@@ -14,10 +15,13 @@ type ClientCfg struct {
 	Timeout time.Duration
 	// PeersRefreshInterval for polling list of peers in etcd.
 	PeersRefreshInterval time.Duration
-	//ConnectionsPerPeer how many gRPC connections to create per grid peer. default: max(1, numCPUs/2).
-	//More connections allow for more messages per second by increase the number of
-	//filehandles used.
+	// ConnectionsPerPeer how many gRPC connections to create per
+	// grid peer, default is max(1, numCPUs/2). More connections
+	// allow for more messages per second by increase the number
```
Comment (epsniff) [inline grid.v3/cfg.go]: damn my typo `by increases` > `but it increases the number...`

Comment (mdmarek) [inline grid.v3/cfg.go]: oh, let me fix..

File: grid.v3/server.go [hunk 2]
```diff
@@ -94,6 +90,11 @@ func (s *Server) Serve(lis net.Listener) error {
 	s.registry.Timeout = s.cfg.Timeout
 	s.registry.LeaseDuration = s.cfg.LeaseDuration
 
+	// Set registry logger.
+	if s.cfg.Logger != nil {
```
Comment (epsniff) [inline grid.v3/server.go]: what is logger is nil?  panic if we try to use it later?

File: grid.v3/registry/registry.go [hunk 3]
```diff
@@ -461,6 +488,12 @@ func (rr *Registry) Deregister(c context.Context, key string) error {
 	return nil
 }
 
+func (rr *Registry) logf(format string, v ...interface{}) {
+	if rr.Logger != nil {
+		rr.Logger.Printf(format, v...)
```
Comment (epsniff) [inline grid.v3/registry/registry.go]: NM I see it.

File: grid.v3/server.go (hunk 2 above)
Comment (epsniff) [inline grid.v3/server.go]: NM found it.

**Pull request merged**